### PR TITLE
Fixed terminating pod check to handle oc 3.11.154+ and pods that have…

### PR DESCRIPTION
**Pull request by superfunhappytime**
… failed termination due to deadline exceeding

### CODE REVIEW

**Comment by superfunhappytime** [comment]
[test]

File: scripts/monitoring/cron-send-terminating-pod-check.py [hunk 1]
```diff
@@ -91,20 +91,33 @@ def check_terminating_pod(pods):
...
-                if item['type'] == "Ready":
-                    time_terminated = item['lastTransitionTime']
+            if 'conditions' in yaml_result['status']:
+                status_condition = yaml_result['status']['conditions']
+                for item in status_condition:
+                    if item['type'] == "Ready":
+                        time_terminated = item['lastTransitionTime']
+            elif ('phase' in yaml_result['status'] and
+                  yaml_result['status']['phase'] == "Failed" and
+                  yaml_result['status']['reason'] == "DeadlineExceeded"):
```
Comment (mrbarge) [inline scripts/monitoring/cron-send-terminating-pod-check.py:111]: If reason isn't in the dict it'll throw a keyerror. Looks like the rest of the new code is very thorough in checking dict keys, so may as well add a check for this one too?

Comment (superfunhappytime) [reply scripts/monitoring/cron-send-terminating-pod-check.py:111]: Good catch, missed that one.

**Review by mrbarge — APPROVED** [review]

**Pull request merged**